base_local_planner/src/obstacle_cost_function.cpp

```diff
@@ -136,9 +137,11 @@ double ObstacleCostFunction::footprintCost (
     return -7.0;
   }
 
-  double occ_cost = std::max(std::max(0.0, footprint_cost), double(costmap->getCost(cell_x, cell_y)));
+  // double occ_cost = std::max(std::max(0.0, footprint_cost), double(costmap->getCost(cell_x, cell_y)));
```

Comment:
Using the max of the footprint causes issues.  Essentially, if the robot's footprint gets within the obstacle 'inscribed radius' inflation (which is lethal if the center of the robot is in it), the cost does not change, even if the robot gets closer to the obstacle.  Using only the center cost (unless the footprint is in collision) allows cost scaling as it gets closer to obstacles.